test cases/common/197 function attributes/meson.build

```diff
@@ -53,6 +53,7 @@ attributes = [
   'used',
   'warn_unused_result',
   'weak',
+  'retain',
```

Comment:
Unconditional check is probably a bad idea, see e.g. later on in the file we do:

```meson
  if c.get_id() == 'gcc' and c.version().version_compare('>= 7.0.0')
    attributes += 'fallthrough'
  endif
```

CI fails because this attribute is lacking and triggers an assert on too-old GCC.

Reply:
Scoped to clang 13+, I'll add a comment about why when rebasing